packages/mobile/src/components/app-navigator/BottomTabNavigator.tsx

```diff
@@ -52,24 +54,35 @@ const createStackScreen = <StackParamList extends ParamListBase>(
   ) => React.ReactNode
 ) => {
   const Stack = createStackNavigator<StackParamList>()
-  return () => (
-    <Stack.Navigator
-      screenOptions={{
-        cardOverlayEnabled: true,
-        cardStyleInterpolator: CardStyleInterpolators.forHorizontalIOS,
-        gestureEnabled: true,
-        gestureResponseDistance: 1000,
-        header: props => <TopBar {...props} />,
-        headerStyle: { height: 87 },
-        headerMode: 'float'
-      }}
-    >
-      {baseScreen(Stack)}
-      <Stack.Screen name='track' component={TrackScreen} />
-      <Stack.Screen name='profile' component={ProfileScreen} />
-      <Stack.Screen name='EditProfile' component={EditProfileScreen} />
-    </Stack.Navigator>
-  )
+  return () => {
+    const dispatchWeb = useDispatchWeb()
```

Comment:
can we name this as a component by chance? not sure what the best perf option is, since its not ideal returning newly created components in each function call. maybe we make a component outside that accepts a Stack prop?

Reply:
Yeah good idea, let me update that